acctz/acctz.proto

```diff
@@ -99,34 +99,30 @@ message SessionInfo {
 // Details of authentication - for gRPC or shell/vendor-CLI.
 message AuthDetail {
   // Identity string is used to identify the user that executed the
-  // command. For instance, it could be the spiffe-id in the case of
+  // command. For instance, it could be the SPIFFE-ID in the case of
   // gRPC or unix-style user-name in the case of shell/vendor-CLI.
   string identity = 1;
 
-  // Privilege level configured on the system.
-  uint32 privilege_level = 2;
-
-  // authentication status
+  // Authentication status
   enum AuthenStatus {
     AUTHEN_STATUS_UNSPECIFIED = 0;
     AUTHEN_STATUS_PERMIT = 1;
     AUTHEN_STATUS_DENY = 2;
   }
-  AuthenStatus status = 3;
+  AuthenStatus status = 2;
 
   // In case of STATUS_DENY, cause for the deny
-  string deny_cause = 4;
+  string deny_cause = 3;
 }
 
 // Command details for shell/vendor-CLI
 message CommandService {
   enum CmdServiceType {
     CMD_SERVICE_TYPE_UNSPECIFIED = 0;
-    CMD_SERVICE_TYPE_SHELL = 1;
```

Comment:
Underlying "shell" accounting to my knowledge is not implemented across any shipping network implementation today so I guess I would like to bring this one back to "how" this would be suggested, what this might entail first before adding a variant here which cascades to the remainder of other messages/field options here.

Authorization and accounting of supported NBI is very well supported but since most systems implement on top of some alternate base OS, the underlying OS shell capabilities (either control or forwarding plane) are not subject to the same AAA.  Generally this stops at if a user can access the shell or not coupled with an underlying group mapping that may or may not allow privilege escalation (su/sudo).

Each implementation could support a number of underlying shells (csh, sh, bash, zsh, etc..) and while there are techniques to "intercept" and log commands via shell profiles, this could likely be overridden by a user and would also call for those subsystems to be able to hook into accounting subsystems outside of general logging (file/syslog)

I fully comprehend that this means that there is lack of visibility into what a user that has these privileges may do in an underlying shell but maybe we can bring this back to if or how this is done in the compute space first and then reintroduce capabilities back into gNSI once defined.

Accounting is also the only service currently concerned with the underlying shell.  Authorization would be an entirely other conversation as well.

Reply:
None of that means that this accounting is not useful or could the issues with the method could not be overcome.  I do not see a reason to remove this.

Reply:
The reason why I would suggest removal and trimming such behaviors is because its currently an undefined wish list item is all.  A protobuf IDL with fields and comments is not the same as a specification that considers how to implement and will result in misperception and ambiguity and thus differences and partial support among implementations.

At a high level, I do not disagree to the usefulness because without such, there is loss of visibility into some actions that are performed by already trusted classes of users however...

A network element is mostly a distributed system wrapped up to abstract a single management interface - there are fully supported interfaces into the network element and then there can be many non-supported interfaces/debug shells such as:

* Underlying controller card OS shells
* Linecard OS/ukern shells
* Other components, SoC - BCRM debug shell, etc...

In that means there are various methods to essentially "jump" or "attach" to such endpoints by breaking out of the supported interface - it's many "systems" internally.  Some but not all capabilities are brought up to the fully supported interfaces (e.g. common and absolutely necessary often providing some abstraction) - these generally inherit the supported AAA infra.

Each one of these interfaces has vastly different underlying resources and may or may not be modifiable from implementation to implementation so I'd like to peel this one back a bit to what this might look like and what might be realistic when we're talking about accounting to various (sub)systems which also as I mentioned would also really apply to any vision of authorization as well.  This problem statement need not apply to network elements either and would pose a similar issue in various compute host scenarios as well.

So agree that it's useful but would start at dissecting the problem statement w/ various scenarios and then talk solutions before adding it to the API definition as that's the easier of all above.

Reply:
My point is that it exists today, though it has the caveat that you mention. But the caveat exists not be cause it is not possible to avoid, but due to the laziness of the implementation.

Reply:
Could you clarify the last statement?  Are you saying command accounting (and/or authorization) for all of the underlying shells/subsystems specified above does exist today?  I had asked in the initial comment if or how this might be done in the compute space because the analogy can be drawn to any OS and not just captive shells.

It seems that this can be overcorrected and solved by not giving access to the subsystems by any public API (CLI in this bucket as well) but that would be counterproductive in reality for everyone.

IMO its a problem statement that needs brainstorming and collaboration on solutions and not just a field in an API declaration

The intent of the PR is to trim undefined behaviors (which this is) with possible reintroduction back at a later point

Reply:
"for all of the underlying shells/subsystems specified above does exist today?"

yes, for one platform we use, it does...

Is the problem you (ebben) see:
   "Today no platform will be able to account for 100% of user actions, so having this toggle in proto is not helpful"
   (this is your aspirational comment I think?)
   
I suppose that's a fine argument, but what if we open FR's with all vendors we deal with to say:

  "Hey, all user input on the cli MUST be accounted, full stop."

yup, we'll be waiting a while but.. that really is the long term goal, I think.
we COULD also just go add the enum / etc at the time that the features start to be realized.

Reply:
These non-command-like services should be split into a different (new) service-type.